kb/server/cache/TransactionCache.java

```diff
@@ -154,7 +155,7 @@ public void remove(Concept concept) {
             Attribute<?> attr = concept.asAttribute();
             // this is probably slower than reading index from vertex but we don't have access to AttributeImpl here (cyclic dep)
             Label attrLabel = attr.type().label();
-            String attrIndex = Schema.generateAttributeIndex(attrLabel, attr.value().toString());
+            String attrIndex = Schema.generateAttributeIndex(attrLabel, AttributeValueConverter.tryConvert(attr.type(), attr.value()).toString());
```

Comment:
here too

Reply:
Yeah this one is definitely unnecessary on second thoughts as the concept was loaded from the KG.

Reply:
Reverted